src/Test/Utilities/Desktop/CommonDiagnosticAnalyzers.cs

```diff
@@ -844,6 +844,40 @@ private void ReportDiagnosticsCore(Action<Diagnostic> addDiagnostic, Location lo
         }
 
         [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+        public class GeneratedCodeAnalyzer2 : DiagnosticAnalyzer
+        {
+            public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                "GeneratedCodeAnalyzer2Warning",
+                "Title",
+                "GeneratedCodeAnalyzer2Message for '{0}'",
```

Comment:
I would suggest to pass the total number of types analyzed (`namedTypes.Count`) as a parameter here. Then, in the test we can prove that both types were visited by the analyzer, but only one gets reported.

Other than that, LGTM.